lib/charms/prometheus_k8s/v1/prometheus.py

```diff
@@ -279,30 +231,79 @@ def jobs(self):
             its scrape targets.
         """
         scrape_jobs = []
-        for job in self._stored.jobs.values():
-            scrape_jobs.append(json.loads(job))
+
+        for relation in self._charm.model.relations[self._relation_name]:
+            job = self._static_scrape_config(relation)
+            if job:
+                scrape_jobs.append(job)
 
         return scrape_jobs
 
+    def _static_scrape_config(self, relation):
+        """Generate the static scrape configuration for a relation.
 
-class PrometheusConsumer(ConsumerBase):
-    _stored = StoredState()
+        Args:
+            relation: an `ops.model.Relation` object whose static
+                scrape configuration is required.
 
-    def __init__(self, charm, name, consumes, multi=False):
-        """Construct a Prometheus charm client.
+        Returns:
+            A static scrape configuration for a specific relation.
+        """
+        if len(relation.units) == 0:
+            return {}
```

Comment:
just wondering if an empty dict is consistent with `return None` and `return scrape_config` below

Reply:
In the sense that they both evaluate to `False`. It would be nice to see type hinting for the return type, but eh